### ISSUE TITLE
[BUG] Fix Publish workflow

### ISSUE THREAD

**Issue by jkaltoft** [kirbydesign/designsystem#1947]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug

 

Publish workflow runs even when it shouldn't.

When a pull request is closed without merging the Publish workflow still runs. It shouldn't. The step where version number in `package.json` is bumped fails, which causes the next actual pull request to fail publishing a new package, because a package with that version number already exists. However, the version number in `package.json` _is_ updated, even though no package is published! The following run of the Publish workflow (next actual pull request) seems to be alright, but it will contain the changes from the failing PRs.

See the failing run of the workflow: https://github.com/kirbydesign/designsystem/actions/runs/1592742519

## Describe how to reproduce the bug

1. Create a PR
2. Target master branch
3. Close PR without merging
4. See error: Publish workflow (partially) runs anyway

## Which Kirby version was used?

v5.0.3

## What was the expected behavior?

When a pull request is closed _without merging_ the Publish workflow should _not_ run and no packages should be published.

## Please complete the following information:
- OS: [e.g. Windows, MacOS, iOS]
- Browser [e.g. Chrome, Safari]
- Browser version [e.g. 22]

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [x] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/master/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [x] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [x] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [x] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [x] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [x] Create a pull-request.

**Comment by jkaltoft** [comment]
The bug _was_ fixed. Unfortunately the solution introduced a new bug that made the publish and the build workflows run over and over again. Probably being triggered by each other (run on push to master branch).

**Comment by jkaltoft** [comment]
Until we have a solution (that doesn't introduce new bugs) we should avoid closing pull requests without merging. If we have to close a PR without merging, then change the target branch to something other than `master` first.

**Comment by jkaltoft** [comment]
The latest fix also had to be reverted: https://github.com/kirbydesign/designsystem/pull/1971

A proper fix is postponed and should be addressed when we handle release management.
